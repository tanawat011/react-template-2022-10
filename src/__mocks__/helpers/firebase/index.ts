import type * as Firebase from 'helpers/firebase'

const firebase = jest.createMockFromModule<typeof Firebase>('helpers/firebase')

module.exports = firebase
