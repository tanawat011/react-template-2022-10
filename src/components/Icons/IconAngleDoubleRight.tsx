type Prop = {
  width?: number
  height?: number
  color?: string
}

export const IconAngleDoubleRight: React.FC<Prop> = ({
  width = 16,
  height = 12,
  color = '#626672',
}) => {
  return (
    <svg
      width={`${width}`}
      height={`${height}`}
      viewBox={`0 0 ${width} ${height}`}
      fill='none'
      xmlns='http://www.w3.org/2000/svg'
    >
      <path
        d='M8.01004 6.6373L2.55093 11.7357C2.17361 12.0881 1.56347 12.0881 1.19017 11.7357L0.28299 10.8885C-0.0943302 10.5361 -0.0943302 9.96626 0.28299 9.61762L4.15253 6.00375L0.28299 2.38988C-0.0943302 2.03749 -0.0943302 1.46767 0.28299 1.11903L1.18615 0.264292C1.56347 -0.0880975 2.17361 -0.0880975 2.54691 0.264292L8.00602 5.3627C8.38736 5.71509 8.38736 6.28491 8.01004 6.6373V6.6373ZM15.717 5.3627L10.2579 0.264292C9.88058 -0.0880975 9.27045 -0.0880975 8.89714 0.264292L7.98997 1.11153C7.61264 1.46392 7.61264 2.03374 7.98997 2.38238L11.8595 5.99625L7.98997 9.61012C7.61264 9.96251 7.61264 10.5323 7.98997 10.881L8.89714 11.7282C9.27446 12.0806 9.8846 12.0806 10.2579 11.7282L15.717 6.6298C16.0943 6.28491 16.0943 5.71509 15.717 5.3627Z'
        fill={color}
      />
    </svg>
  )
}
