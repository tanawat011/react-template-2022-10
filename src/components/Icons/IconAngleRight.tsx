type Prop = {
  width?: number
  height?: number
  color?: string
}

export const IconAngleRight: React.FC<Prop> = ({ width = 8, height = 12, color = '#626672' }) => {
  return (
    <svg
      width={`${width}`}
      height={`${height}`}
      viewBox={`0 0 ${width} ${height}`}
      fill='none'
      xmlns='http://www.w3.org/2000/svg'
    >
      <path
        d='M7.72747 6.6373L2.46094 11.7357C2.09693 12.0881 1.50832 12.0881 1.14818 11.7357L0.273008 10.8885C-0.0910026 10.5361 -0.0910026 9.96626 0.273008 9.61762L4.00605 6.00375L0.273008 2.38988C-0.0910026 2.03749 -0.0910026 1.46767 0.273008 1.11903L1.14431 0.264292C1.50832 -0.0880974 2.09693 -0.0880974 2.45707 0.264292L7.7236 5.3627C8.09148 5.71509 8.09148 6.28491 7.72747 6.6373Z'
        fill={color}
      />
    </svg>
  )
}
