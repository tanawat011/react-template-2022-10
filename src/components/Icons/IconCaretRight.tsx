type Prop = {
  width?: number
  height?: number
  color?: string
}

export const IconCaretRight: React.FC<Prop> = ({ width = 8, height = 12, color = '#626672' }) => {
  return (
    <svg
      width={`${width}`}
      height={`${height}`}
      viewBox={`0 0 ${width} ${height}`}
      fill='none'
      xmlns='http://www.w3.org/2000/svg'
    >
      <path
        d='M0 11.1914V0.80859C0 0.0896493 1.02183 -0.270386 1.61943 0.237973L7.72217 5.42937C8.09261 5.7445 8.09261 6.25548 7.72217 6.57061L1.61943 11.762C1.02183 12.2704 0 11.9103 0 11.1914Z'
        fill={color}
      />
    </svg>
  )
}
