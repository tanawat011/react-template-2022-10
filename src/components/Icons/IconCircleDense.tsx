type Prop = {
  width?: number
  height?: number
  color?: string
}

export const IconCircleDense: React.FC<Prop> = ({ width = 12, height = 12, color = '#626672' }) => {
  return (
    <svg
      width={`${width}`}
      height={`${height}`}
      viewBox={`0 0 ${width} ${height}`}
      fill='none'
      xmlns='http://www.w3.org/2000/svg'
    >
      <circle cx='6' cy='6' r='6' fill={color} />
    </svg>
  )
}
