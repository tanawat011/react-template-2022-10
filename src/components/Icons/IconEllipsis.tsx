type Prop = {
  width?: number
  height?: number
  color?: string
}

export const IconEllipsis: React.FC<Prop> = ({ width = 4, height = 14, color = '#626672' }) => {
  return (
    <svg
      width={`${width}`}
      height={`${height}`}
      viewBox={`0 0 ${width} ${height}`}
      fill='none'
      xmlns='http://www.w3.org/2000/svg'
    >
      <path
        d='M2 4.96774C3.10556 4.96774 4 5.87661 4 7C4 8.12339 3.10556 9.03226 2 9.03226C0.894444 9.03226 0 8.12339 0 7C0 5.87661 0.894444 4.96774 2 4.96774ZM0 2.03226C0 3.15565 0.894444 4.06452 2 4.06452C3.10556 4.06452 4 3.15565 4 2.03226C4 0.908871 3.10556 0 2 0C0.894444 0 0 0.908871 0 2.03226ZM0 11.9677C0 13.0911 0.894444 14 2 14C3.10556 14 4 13.0911 4 11.9677C4 10.8444 3.10556 9.93548 2 9.93548C0.894444 9.93548 0 10.8444 0 11.9677Z'
        fill={color}
      />
    </svg>
  )
}
