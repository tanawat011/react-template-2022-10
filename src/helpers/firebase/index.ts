export * from './deleteCollection'
export * from './deleteDocument'
export * from './getAllDocuments'
export * from './getDocument'
export * from './getDocumentWithRef'
export * from './setDocument'
export * from './subscribeCollection'
export * from './subscribeDocument'
export * from './updateDocument'
