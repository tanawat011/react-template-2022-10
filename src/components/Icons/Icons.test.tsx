import { renderWithProviders } from 'helpers/test'

import { IconAngleDoubleRight } from './IconAngleDoubleRight'
import { IconAngleRight } from './IconAngleRight'
import { IconCaretRight } from './IconCaretRight'
import { IconCheck } from './IconCheck'
import { IconCircleDense } from './IconCircleDense'
import { IconEllipsis } from './IconEllipsis'

describe('<Icon />', () => {
  test('<IconAngleDoubleRight />, render snapshot', () => {
    const { asFragment } = renderWithProviders(<IconAngleDoubleRight />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('<IconAngleRight />, render snapshot', () => {
    const { asFragment } = renderWithProviders(<IconAngleRight />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('<IconCaretRight />, render snapshot', () => {
    const { asFragment } = renderWithProviders(<IconCaretRight />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('<IconCheck />, render snapshot', () => {
    const { asFragment } = renderWithProviders(<IconCheck />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('<IconCircleDense />, render snapshot', () => {
    const { asFragment } = renderWithProviders(<IconCircleDense />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('<IconEllipsis />, render snapshot', () => {
    const { asFragment } = renderWithProviders(<IconEllipsis />)

    expect(asFragment()).toMatchSnapshot()
  })
})
