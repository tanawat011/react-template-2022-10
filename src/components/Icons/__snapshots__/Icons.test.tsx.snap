// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> <IconAngleDoubleRight />, render snapshot 1`] = `
<DocumentFragment>
  <svg
    fill="none"
    height="12"
    viewBox="0 0 16 12"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.01004 6.6373L2.55093 11.7357C2.17361 12.0881 1.56347 12.0881 1.19017 11.7357L0.28299 10.8885C-0.0943302 10.5361 -0.0943302 9.96626 0.28299 9.61762L4.15253 6.00375L0.28299 2.38988C-0.0943302 2.03749 -0.0943302 1.46767 0.28299 1.11903L1.18615 0.264292C1.56347 -0.0880975 2.17361 -0.0880975 2.54691 0.264292L8.00602 5.3627C8.38736 5.71509 8.38736 6.28491 8.01004 6.6373V6.6373ZM15.717 5.3627L10.2579 0.264292C9.88058 -0.0880975 9.27045 -0.0880975 8.89714 0.264292L7.98997 1.11153C7.61264 1.46392 7.61264 2.03374 7.98997 2.38238L11.8595 5.99625L7.98997 9.61012C7.61264 9.96251 7.61264 10.5323 7.98997 10.881L8.89714 11.7282C9.27446 12.0806 9.8846 12.0806 10.2579 11.7282L15.717 6.6298C16.0943 6.28491 16.0943 5.71509 15.717 5.3627Z"
      fill="#626672"
    />
  </svg>
</DocumentFragment>
`;

exports[`<Icon /> <IconAngleRight />, render snapshot 1`] = `
<DocumentFragment>
  <svg
    fill="none"
    height="12"
    viewBox="0 0 8 12"
    width="8"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7.72747 6.6373L2.46094 11.7357C2.09693 12.0881 1.50832 12.0881 1.14818 11.7357L0.273008 10.8885C-0.0910026 10.5361 -0.0910026 9.96626 0.273008 9.61762L4.00605 6.00375L0.273008 2.38988C-0.0910026 2.03749 -0.0910026 1.46767 0.273008 1.11903L1.14431 0.264292C1.50832 -0.0880974 2.09693 -0.0880974 2.45707 0.264292L7.7236 5.3627C8.09148 5.71509 8.09148 6.28491 7.72747 6.6373Z"
      fill="#626672"
    />
  </svg>
</DocumentFragment>
`;

exports[`<Icon /> <IconCaretRight />, render snapshot 1`] = `
<DocumentFragment>
  <svg
    fill="none"
    height="12"
    viewBox="0 0 8 12"
    width="8"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 11.1914V0.80859C0 0.0896493 1.02183 -0.270386 1.61943 0.237973L7.72217 5.42937C8.09261 5.7445 8.09261 6.25548 7.72217 6.57061L1.61943 11.762C1.02183 12.2704 0 11.9103 0 11.1914Z"
      fill="#626672"
    />
  </svg>
</DocumentFragment>
`;

exports[`<Icon /> <IconCheck />, render snapshot 1`] = `
<DocumentFragment>
  <svg
    fill="none"
    height="12"
    viewBox="0 0 16 12"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M5.43431 11.7643L0.234305 6.53442C-0.0781016 6.22021 -0.0781016 5.71077 0.234305 5.39654L1.36565 4.25865C1.67806 3.94442 2.18462 3.94442 2.49703 4.25865L6 7.78174L13.503 0.235652C13.8154 -0.0785507 14.3219 -0.0785507 14.6344 0.235652L15.7657 1.37353C16.0781 1.68774 16.0781 2.19718 15.7657 2.51141L6.56569 11.7644C6.25325 12.0786 5.74672 12.0786 5.43431 11.7643V11.7643Z"
      fill="#626672"
    />
  </svg>
</DocumentFragment>
`;

exports[`<Icon /> <IconCircleDense />, render snapshot 1`] = `
<DocumentFragment>
  <svg
    fill="none"
    height="12"
    viewBox="0 0 12 12"
    width="12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="6"
      cy="6"
      fill="#626672"
      r="6"
    />
  </svg>
</DocumentFragment>
`;

exports[`<Icon /> <IconEllipsis />, render snapshot 1`] = `
<DocumentFragment>
  <svg
    fill="none"
    height="14"
    viewBox="0 0 4 14"
    width="4"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2 4.96774C3.10556 4.96774 4 5.87661 4 7C4 8.12339 3.10556 9.03226 2 9.03226C0.894444 9.03226 0 8.12339 0 7C0 5.87661 0.894444 4.96774 2 4.96774ZM0 2.03226C0 3.15565 0.894444 4.06452 2 4.06452C3.10556 4.06452 4 3.15565 4 2.03226C4 0.908871 3.10556 0 2 0C0.894444 0 0 0.908871 0 2.03226ZM0 11.9677C0 13.0911 0.894444 14 2 14C3.10556 14 4 13.0911 4 11.9677C4 10.8444 3.10556 9.93548 2 9.93548C0.894444 9.93548 0 10.8444 0 11.9677Z"
      fill="#626672"
    />
  </svg>
</DocumentFragment>
`;
